### ISSUE TITLE
possible issue displaying double text (preact-live)

### ISSUE THREAD

**Issue by jeremy-coleman** [developit/preact#1533]
i don't think it's a 'bug', likely my fault but it'd be cool to get the preact-live live component up and running

![image](https://user-images.githubusercontent.com/25135855/56046244-d2dbf080-5d10-11e9-98e2-2f2469f4cc98.png)

code is here, its probably something to do with className usage , onChange handler, or event target. 
https://github.com/jeremy-coleman/preact-x-serve/tree/master/src/preact-live

the text area displays double text, also compared to the react version, the default code after an arrow is wrapped in enclosing parenthesis  (which comes from a string in the source code, and just magically appears), causing it not to render without manually updating first. the () => div xxx at the top is there for convenience to copy/paste into the text area

**Comment by marvinhagemeister** [comment]
I cloned the repo and did the following steps:

1. `npm install`
2. `npm start`
2.1. Error: `sh: 1: cross-env: not found`, can be fixed via `npm i -D cross-env`
2.2. Another error:
```
 ERROR in ./src/index.tsx
  Module not found: Error: Can't resolve './Home' in '/home/marvinh/dev/test/preact-bugs/preact-x-serve/src'
```
Seems to be caused by inconsistent casing of filenames and imports. The actual filename is `home.tsx` with a lowercase `h`, but the import is in uppercase. This usually happens when working on a Mac where the filesystem is case insensitive. Same error with `errorBoundary` vs `ErrorBoundary`

Now that we got it running we can start looking for the bug :+1:

After some digging the issue is caused by having 2 DOM Elements on top of each other:

```html
<div>
  <!-- Actual input where the user will type in. Text color is #444 -->
  <textarea></textarea>
  <!-- This elment is a tokenized version for syntax highlighting etc -->
  <pre>...</pre>
</div>
```

So how do other editors like `Monaco` which is the base for `vscode` and `codesandbox` do it? The fix is rather simple and they simply make the text from the `textarea` invisible:

```css
textarea {
  color: transparent;
  background: transparent;
}
```

The downside to this is that you'll lose the text cursor. For that they create a special `div` that they position manually. My guess is that they know exactly where the cursor is and can place it at the exact coordinates of the tokenised version. Same trick needs to be applied for text selection. By default the highlight color is `white` leading to the highlighted text to bleed through. For that there is a handy [::selection](https://developer.mozilla.org/en-US/docs/Web/CSS/::selection) selector in CSS.

Hope this helps :+1:
